models/eligibility_check.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * An EligibilityCheck is a request from the User to validate his/her DSU (dichiarazione sostitutiva unica). It might be referred also as DSU Request.
+ * An EligibilityCheck is identified by the User's unique identifier (thus at most one EligibilityCheck can exist for a User at a given time).
+ * It can be a case of Success of Failure: in the first case it will also include a list of the family members of the associated DSU
+ */
+
+import * as DocumentDb from "documentdb";
+import { Either, left, right } from "fp-ts/lib/Either";
+import * as DocumentDbUtils from "io-functions-commons/dist/src/utils/documentdb";
+import { DocumentDbModel } from "io-functions-commons/dist/src/utils/documentdb_model";
+import * as t from "io-ts";
+import { pick, tag } from "italia-ts-commons/lib/types";
+import { EligibilityCheck } from "../generated/definitions/EligibilityCheck";
+import { EligibilityCheckFailure } from "../generated/definitions/EligibilityCheckFailure";
+import { EligibilityCheckSuccess } from "../generated/definitions/EligibilityCheckSuccess";
+import { keys } from "../utils/types";
+
+export const ELIGIBILITY_CHECK_COLLECTION_NAME = "eligibility-checks";
+export const ELIGIBILITY_CHECK_MODEL_PK_FIELD = "id";
+
+interface IRetrievedEligibilityCheck {
+  readonly kind: "IRetrievedEligibilityCheck";
+}
+export const RetrievedEligibilityCheck = tag<IRetrievedEligibilityCheck>()(
+  t.intersection([EligibilityCheck, DocumentDbUtils.RetrievedDocument])
+);
+export type RetrievedEligibilityCheck = t.TypeOf<
+  typeof RetrievedEligibilityCheck
+>;
+
+interface INewEligibilityCheckTag {
+  readonly kind: "INewEligibilityCheck";
+}
+export const NewEligibilityCheck = tag<INewEligibilityCheckTag>()(
+  t.intersection([EligibilityCheck, DocumentDbUtils.NewDocument])
```

Comment:
you can't use the generated (API) types, model fields are camelCased

Reply:
there's no guarantees that API types ad model (db) types are the same, you have to provide some mapping. see https://github.com/pagopa/io-functions-commons/blob/master/src/models/profile.ts#L41 (or other models in io-fn-commons)

Reply:
and https://github.com/pagopa/io-functions-app/blob/b16c6d3c34948b40ac8384408bb0920365f0b53e/utils/profiles.ts#L40